.github/workflows/rust.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  fmt:
+    name: Formatting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          components: rustfmt
+          profile: minimal
+          override: true
+      - name: Run rustfmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          components: clippy
+          profile: minimal
+          override: true
```

Comment:
`nightly` is needed because there is a transitive dependency `io-lifetimes` that uses a feature `rustc_attrs` which may not be used on stable rust. 
```
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/crocus/.cargo/registry/src/github.com-1ecc6299db9ec823/io-lifetimes-0.5.3/src/lib.rs:31:26
   |
31 | #![cfg_attr(rustc_attrs, feature(rustc_attrs))]
   |  
```
I agree that we should define the rust version in the CI script file.

Reply:
Hmm, I can run clippy locally using stable Rust version 1.65. What command are you using to get this error?

Reply:
Hm. I just run `cargo clippy -- -D clippy::all`. I will check it out once again. Maybe the reason is my rust versioning mess. Thanks for advance.

Reply:
Yeah. It's my fault. `cargo clean` execution could help sometimes 😄